core/granite/commons/versions.ts

```diff
@@ -0,0 +1,94 @@
+export const MIN_OLLAMA_VERSION = "0.6.5";
+
+export function checkMinimumServerVersion(version: string): boolean {
+  const result = compareVersions(version, MIN_OLLAMA_VERSION) >= 0;
+  return result;
+}
+
+// copied from packages/config-yaml/src/schemas/data/index.ts
+const SEMVER_REGEX =
+  /^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9A-Za-z-][0-9A-Za-z-]*)(?:\.(?:0|[1-9A-Za-z-][0-9A-Za-z-]*))*))?(?:\+([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?$/;
+
+export function isSemVer(version: string | undefined): boolean {
+  return version !== undefined && SEMVER_REGEX.test(version);
+}
+
+/**
+ * Compares two version strings and returns a number indicating their order.
+ *
+ * @param v1 - The first version string to compare.
+ * @param v2 - The second version string to compare.
+ * @returns A number indicating the order of the versions:
+ *          - `1` if `v1` is greater than `v2`
+ *          - `-1` if `v1` is less than `v2`
+ *          - `0` if `v1` is equal to `v2`
+ *
+ * The function first checks if the versions are valid semantic versions (SemVer).
+ * If both versions are SemVer, it compares their major, minor, and patch parts.
+ * If the versions have pre-release identifiers, it compares them as well.
+ * If neither version is SemVer, it returns `0`.
+ */
+export function compareVersions(v1: string, v2: string): number {
```

Comment:
That's a lot of tricky code to have without test cases ... while Continue doesn't directly rely on the 'semver' module, it's transitively pulled in various ways. Is there any reason not to just use that?

Reply:
I initially didn't want to add a dependency to a 30kB lib, but it is indeed already pulled in at runtime just fr that simple check. I verified semver works, so I'll remove this code.